### PR TITLE
Add example for multi-objective functions

### PR DESCRIPTION
**Pull request by ckaldemeyer**
This adds an example for multi-objective optimization within the Python tutorials. It is inspired by other tutorials but pretty short to avoid redundancies.

Unfortunately, I could not check locally if sphinx generates the right output because your configuration seems to differ from the one I am used to. So you either help me with building it on my local system or just check it by yourself quickly ;-)

Let me know if it suits you like this!

### CODE REVIEW

**Comment by bluescarni** [comment]
Thanks for the PR!

I think that currently pagmo/pygmo's docs do not build with the latest version of sphinx, due to some sphinx issues. What I usually do then is to install the previous version of sphinx locally, like this:

```console
pip install --user 'sphinx<1.7'
```

And then build the documentation like this:

```console
make html SPHINXBUILD=~/.local/bin/sphinx-build
```

I'll let @darioizzo comment on the tutorial as he's the algorithm guy, let me just note that there is an indentation error in the definition of the fitness function that's making the CI fail.

**Comment by ckaldemeyer** [comment]
> Thanks for the PR!
> 
> I think that currently pagmo/pygmo's docs do not build with the latest version of sphinx, due to some sphinx issues. What I usually do then is to install the previous version of sphinx locally, like this:
> 
> pip install --user 'sphinx<1.7'
> 
> And then build the documentation like this:
> 
> make html SPHINXBUILD=~/.local/bin/sphinx-build
> 
> I'll let @darioizzo comment on the tutorial as he's the algorithm guy, let me just note that there is an indentation error in the definition of the fitness function that's making the CI fail.

Allright, thanks for the remark! I have solved the indentation error as I have replaced the function with an easier one which also made the numpy import obsolete. I'll then wait for @darioizzo .

**Comment by ckaldemeyer** [comment]
I am wondering why TravisCI is still unhappy with Python 3.6...

**Comment by bluescarni** [comment]
The Python 3.6 build is the one building the docs, so any doc issue will show up there. It looks like it's building fine now though :)

**Comment by darioizzo** [comment]
The Tutorial is

File: doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+*Schaffer, J. David (1984). Some experiments in machine learning using vector
+evaluated genetic algorithms (artificial intelligence, optimization, adaptation,
+pattern recognition) (PhD). Vanderbilt University.* and illustrated
+`here <https://en.wikipedia.org/wiki/Test_functions_for_optimization#Test_functions_for_multi-objective_optimization>`_.
+
+The implementation as UDP can be realized as follows:
+
+.. doctest::
+
+    >>> import pygmo as pg
```
Comment (darioizzo) [inline doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: no need to import pygmo here. udps can be defined also without pygmo available. You can import pygmo later when using the udp to instantiate a pygmo.problem

File: doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    >>> # create population
+    >>> pop = pg.population(prob, size=200)
+    >>> # select algorithm
+    >>> algo = pg.algorithm(pg.nsga2(gen=40))
+    >>> # run optimization
+    >>> pop = algo.evolve(pop)
+    >>> # print results
+    >>> fits, vectors = pop.get_f(), pop.get_x()
+    >>> print(fits[:1]) #doctest: +SKIP
```
Comment (darioizzo) [inline doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: Not sure if printing one fitness says anything on the problem being solved. Maybe here the tutorial could benefit from the extraction of the non dominated front? Or its plot?

Comment (ckaldemeyer) [reply doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: I have printed the NDF. See: https://github.com/esa/pagmo2/pull/191/commits/99206785bfb53f8fe36738190f69fcf57f316d57

File: doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst (hunk 1 above)
Comment (ckaldemeyer) [reply doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: I have moved the import. See: https://github.com/esa/pagmo2/pull/191/commits/99206785bfb53f8fe36738190f69fcf57f316d57

File: doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
+.. _py_tutorial_coding_udp_multi_objective:
+
+Coding a User Defined Problem with constraints multiple objectives
```
Comment (darioizzo) [inline doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: with constraints multiple objectives should be "with multiple objectives" ?

Comment (darioizzo) [reply doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: You can skip the CI for these type of typo fixes by embedding "[SKIP CI]" in your commit message (i think in front)

Comment (ckaldemeyer) [reply doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst]: Done! Weird that I did not notice the headline...

Have you checked the rendering of the mathematical description? I did not build the docs locally...

**Pull request merged**